### ISSUE TITLE
Read filter CIGAR_AGREES_WITH_ALIGNEMENT is broken

### ISSUE THREAD

**Issue by akiezun** [broadinstitute/gatk#348]
`ReadFilterLibrary.CIGAR_AGREES_WITH_ALIGNEMENT` is bogus. The code is:

`read -> read.getReadUnmappedFlag() || read.getAlignmentStart() == -1 || read.getAlignmentStart() == SAMRecord.NO_ALIGNMENT_START || read.getAlignmentBlocks().size() >= 0;` and it will always return true because a java List cannot have size < 0. This is code form the old GATK and it's broken there too. Whatever the intention was for this filter, it's just using CPU not filtering anything.

**Comment by akiezun** [comment]
@droazen how can we know if that interpretation above is correct? Do we need a filter for `read.getAlignmentBlocks().size() > 0`? If yes, please rename the issue to clarify.

**Comment by akiezun** [comment]
assigning @droazen to clarify. feel free to unassign afterwards

**Comment by vdauwera** [comment]
FYI in Classic GATK we are reworking filters so that MalformedReadFilter will centralize all these checks including what was in BadCigar. So cigar sanity checking will be part of basic read format checking. 

Recommend waiting on this and porting when that is done.

**Comment by akiezun** [comment]
is there a ticket number?

**Comment by vdauwera** [comment]
Some work being done in gsa-unstable#930 but the full read filter fusing will probably be a separate ticket. Need to assess status tomorrow. If you want to assign this to me I will ensure this ticket gets updated accordingly.

**Comment by akiezun** [comment]
done

**Comment by akiezun** [comment]
something got merged in https://github.com/broadinstitute/gsa-unstable/pull/930/

@akiezun to investigate what, if anything, needs to be ported

**Comment by droazen** [comment]
@akiezun This read filter (`ReadFilterLibrary.CIGAR_AGREES_WITH_ALIGNEMENT`) no longer exists in `ReadFilterLibrary`. What should be done for this ticket?

**Comment by akiezun** [comment]
not sure but will know for alpha-1
